code-execution-engines/langchain4j-code-execution-engine-v8/src/main/java/dev/langchain4j/code/v8/V8JavaScriptExecutionEngine.java

```diff
@@ -0,0 +1,22 @@
+package dev.langchain4j.code.v8;
+
+import com.caoccao.javet.exceptions.JavetException;
+import com.caoccao.javet.interop.V8Host;
+import com.caoccao.javet.interop.V8Runtime;
+import com.caoccao.javet.values.V8Value;
+import dev.langchain4j.code.CodeExecutionEngine;
+
+public class V8JavaScriptExecutionEngine implements CodeExecutionEngine {
+
+
+    @Override
+    public String execute(String code) {
+        try (V8Runtime v8Runtime = V8Host.getV8Instance().createV8Runtime()) {
```

Comment:
is it expensive to create a runtime? Should it be cached?

Reply:
Thanks for noting that

I indeed profiled it and as I know the library internals, it's kind of heavy each time

loading the native library and make some other stuffs

I used my open source project
https://github.com/omarmahamid/async

which can I profile the application
here what I got

![image](https://github.com/user-attachments/assets/97925b0c-b34f-4f70-9f01-f73ebc797b8a)